test/ConfigTest.php

```diff
@@ -4,86 +4,51 @@
 
 namespace LaminasTest\ServiceManager;
 
+use Laminas\ContainerConfigTest\TestAsset\Delegator1Factory;
 use Laminas\ServiceManager\Config;
-use Laminas\ServiceManager\ServiceManager;
+use Laminas\ServiceManager\ConfigInterface;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use LaminasTest\ServiceManager\TestAsset\AbstractFactoryFoo;
+use LaminasTest\ServiceManager\TestAsset\InvokableObject;
+use LaminasTest\ServiceManager\TestAsset\SimpleInitializer;
 use PHPUnit\Framework\TestCase;
 
 /**
  * @covers \Laminas\ServiceManager\Config
+ * @psalm-import-type ServiceManagerConfigurationType from ConfigInterface
  */
 final class ConfigTest extends TestCase
 {
-    public function testMergeArrays(): void
```

Comment:
removed this feature. The `Config` class is marked `final` now, I have no idea why some1 needs a dedicated config for a plugin manager. If still needed, creating a decorator to re-implement this feature in upstream projects seems to be good enough for me.

/cc @froschdesign I guess that needs to be added to the migration guide?

Reply:
I added this pull request to the related issue report: #178

Reply:
I totally removed `Config` + `ConfigInterface` in my latest work. Its just a useless class/interface anyways since the config could've always been passed to `ServiceManager#__construct`.